src/check.rs

```diff
@@ -106,3 +105,11 @@ fn check_http_url(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
         Err(e) => Err(CheckError::Http(HttpError::Fetch(url.clone(), e))),
     }
 }
+
+/// Perform blocking HTTP HEAD request.
+fn head_request(url: &str) -> Result<reqwest::blocking::Response, reqwest::Error> {
+    reqwest::blocking::Client::builder()
+        .build()?
+        .get(url)
+        .send()
```

Comment:
This is not actually a HEAD request, it's still a GET.
```suggestion
    reqwest::blocking::Client::new().head(url).send()
```
Is there any reason you made this a separate function? I think it was more clear inline.